Mathlib/RingTheory/Localization/Integral.lean

```diff
@@ -466,3 +466,35 @@ lemma isAlgebraic_of_isFractionRing {R S} (K L) [CommRing R] [CommRing S] [Field
     apply IsIntegral.tower_top (R := R)
     apply IsIntegral.map (IsScalarTower.toAlgHom R S L)
     exact Algebra.IsIntegral.isIntegral (s : S)
+
+namespace Polynomial
+
+lemma isLocalization {R} [CommRing R] (S : Submonoid R) (A) [CommRing A] [Algebra R A]
+    [IsLocalization S A] :
+    letI := (mapRingHom (algebraMap R A)).toAlgebra; IsLocalization (S.map C) A[X] := by
+  letI := (mapRingHom (algebraMap R A)).toAlgebra
+  constructor
+  · rintro ⟨_, r, hr, rfl⟩
+    simpa [RingHom.algebraMap_toAlgebra] using (IsLocalization.map_units A ⟨r, hr⟩).map C
+  · intro z
+    obtain ⟨b, hb⟩ := IsLocalization.integerNormalization_spec S z
+    refine ⟨⟨IsLocalization.integerNormalization S z, _, b, b.2, rfl⟩, ?_⟩
+    ext i
+    simp only [RingHom.algebraMap_toAlgebra, coe_mapRingHom, map_C, coeff_mul_C, coeff_map, hb,
+      mul_comm, Algebra.smul_def]
+  · intros x y e
+    rw [← sub_eq_zero, ← map_sub, RingHom.algebraMap_toAlgebra, ← RingHom.mem_ker,
+      Polynomial.ker_mapRingHom, Ideal.mem_map_C_iff] at e
+    simp only [coeff_sub, RingHom.mem_ker, map_sub, sub_eq_zero,
+      IsLocalization.eq_iff_exists S] at e
+    choose c hc using e
+    refine ⟨⟨_, _, ((x.support ∪ y.support).prod c).2, rfl⟩, ?_⟩
```

Comment:
What is going on here mathematically? You're comparing two polynomials that you want to show equal, why do you care about the support? I am asking mainly to see how much pain would be to generalize this to power series and/or MVpolynomials.

Reply:
Should work just the same for power series and mv polynomials, yes. We have a sequence `c` such that `c n * x.coeff n = c n * y.coeff n` for all `n`. Now we want to find a single `c'` such that `c * x = c * y`. Answer: take a common multiple of all the `c n` for which either `x.coeff n` or `y.coeff n` is non-zero.

Reply:
Since this is not the "usual math proof" (to my knowledge), would you mind half-a-line of doc in this proof to make this explicit? I think it can be interesting while inspecting it later on.